static/app/views/dashboardsV2/detail.tsx

```diff
@@ -724,3 +697,49 @@ const StyledPageContent = styled(PageContent)`
 `;
 
 export default withApi(withOrganization(DashboardDetail));
+
+/**
+ * Reads the layout from an array of widgets.
+ */
+function getDashboardLayout(widgets: Widget[]): RGLLayout[] {
+  return widgets
+    .filter(({layout}) => !!layout)
+    .map(({layout, ...widget}) => ({
+      ...(layout as RGLLayout),
+      i: constructGridItemKey(widget),
+    }));
+}
+
+// Keys for grid layout values we track in the server
+const STORE_KEYS = ['x', 'y', 'w', 'h', 'minW', 'maxW', 'minH', 'maxH'];
+
+/**
+ * Creates a new DashboardDetails object with the layouts associated with
+ * widgets for outgoing requests.
+ */
+function constructDashboardWidgetsWithLayout(
+  dashboard: DashboardDetails,
+  layout: RGLLayout[]
+): DashboardDetails {
+  return {
+    ...dashboard,
+    widgets: dashboard.widgets.map(widget => {
+      const matchingLayout = layout.find(({i}) => i === constructGridItemKey(widget));
+      return {...widget, layout: pick(matchingLayout, STORE_KEYS)};
+    }),
+  };
+}
+
+function isLayoutEqual(prevLayouts: RGLLayout[], newLayouts: RGLLayout[]): boolean {
```

Comment:
I was going to go for a simple boolean flag in the `onLayoutChange` handler, but there are issues with it triggering for some reason in Jest tests, and also when changing the screen size to mobile.